src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -557,8 +557,17 @@ private static Exception CreateHttpRequestException(Exception inner = null)
             return new HttpRequestException(SR.net_http_client_execution_error, inner);
         }
 
+        private static void ResponseFormatAssert(bool condition)
+        {
+            if (!condition)
+            {
+                throw new HttpRequestException(SR.net_http_unix_invalid_response);
```

Comment:
Using a ```new HttpRequestException``` instead of ```CreateHttpRequestException``` to maintain parity with ```WinHttpHandler``` w.r.t the message of the exception thrown.